OpenRA.Mods.AS/Traits/Nydus.cs

```diff
@@ -0,0 +1,207 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2015- OpenRA.Mods.AS Developers (see AUTHORS)
+ * This file is a part of a third-party plugin for OpenRA, which is
+ * free software. It is made available to you under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation. For more information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Mods.AS.Activities;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.AS.Traits
+{
+	[Desc("This player actor has this many Nydus canals. Must be attached to the player by attaching in player.yaml, otherwise Nydus logic won't work.")]
+	public class NydusCounterInfo : TraitInfo<NydusCounter> { }
+	public class NydusCounter
+	{
+		public int Count = 0;
+		public Actor PrimaryActor = null;
+	}
+
+	[Desc("This actor can teleport actors like Nydus canels in SC1. Assuming static object. The actor must have PrimaryBuilding trait so that exit can be specified!")]
+	public class NydusInfo : ITraitInfo
```

Comment:
This should implement IRulesetLoaded and then check the Player actor if it has the neccessary NydusCounter trait and crash appropriately if not.